src/Marten.CommandLine/Commands/Projection/IProjectionHost.cs

```diff
@@ -14,7 +16,7 @@ public interface IProjectionHost
 {
     IReadOnlyList<IProjectionStore> AllStores();
     void ListenForUserTriggeredExit();
-    Task<RebuildStatus> TryRebuildShards(IProjectionDatabase database, IReadOnlyList<AsyncProjectionShard> asyncProjectionShards);
+    Task<RebuildStatus> TryRebuildShards(IProjectionDatabase database, IReadOnlyList<AsyncProjectionShard> asyncProjectionShards, TimeSpan? shardTimeout=null);
```

Comment:
Ugh, I'm sorry, but we've gotta do that as an overload, not as an optional argument. That's actually a breaking change that would require a code compilation, even though the code itself wouldn't have to change